src/components/CountryDetails.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+import { Link } from 'react-router-dom'
+
+export default class CountryDetails extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            country: {
+                name: '',
+                capital: '',
+                area: 0,
+                borders:[]
+            },
+            code: ''
+        }
+    }
+
+    getData() {
+        axios.get(`https://restcountries.eu/rest/v2/alpha/${this.props.match.params.id}`)
+            .then(({data}) => this.setState({
```

Comment:
Si vas a hacer uso del estado en el setState pasale un callback

`this.setState(state => ({...state}))`

Reply:
Cierto! Gracias. Me lo apunto para hacer refactor.